File: backend/rs-client/java/src/main/java/com/foros/rs/client/service/AdvertisingChannelService.java
package com.foros.rs.client.service;

import com.foros.rs.client.RsClient;
import com.foros.rs.client.model.advertising.channel.Channel;
import com.foros.rs.client.model.advertising.channel.ChannelSelector;

public class AdvertisingChannelService extends EntityServiceSupport<Channel, ChannelSelector> {

    public AdvertisingChannelService(RsClient rsClient) {
        super(rsClient, "/channels/advertising");
    }
}

File: backend/rs-client/php/src/service/AdvertisingChannelService.php
<?php

/**
 
 */
class AdvertisingChannelService {

    private $configuration;

    public function __construct(RsConfiguration $configuration) {
        $this->configuration = $configuration;
    }

    
    public function get (array $channelSelector) {
        $client = $this->configuration->createClient();
        $parameters = null;
        $data = null;
    
        $parameters = $channelSelector;
        
    
        return $client
                ->request("GET" ,"/channels/advertising/", $parameters, $data);
    }
    
    public function perform ($channelOperations) {
        $client = $this->configuration->createClient();
        $parameters = null;
        $data = null;
    
        
        $data = $channelOperations;
    
        return $client
                ->request("POST" ,"/channels/advertising/", $parameters, $data);
    }
}
